### DISCUSSION TITLE
Experimental feature: Limit the number of batched tasks

### DISCUSSION THREAD

**Discussion by Kerollmops** [meilisearch/product#713] [Experimental features]
<table>
    <thead>
        <tr>
            <th>Feature name</th>
  			<th>How to enable</th>
  			<th>Description</th>
  			<th>Missing for stabilization</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>limit the number of batched tasks</td>
            <td>

You can export the env variable, the command line argument, or the config file to limit Meilisearch to processing only a given number of tasks simultaneously.
```bash
# You can either use this environment variable
MEILI_EXPERIMENTAL_MAX_NUMBER_OF_BATCHED_TASKS=100

# directly use this CLI argument
--experimental-max-number-of-batched-tasks 100

# Or via the config file
experimental_max_number_of_batched_tasks = 100
```
  
 </td>
  			<td>Reduces the size of the batched tasks to reduce the RAM usage of the engine</td>
  			<td>We could remove this flag and make the engine sufficiently good to do choose the right number of tasks to process simultaneously</td>
        </tr>
    </tbody>
</table>

### How to use the feature?

1. Run Meilisearch with the environment variable or the command line argument flag described above
2. Profit.

## Why is this feature not stable yet?

We are confident that this parameter will become useless in a short time. We are exposing this option to help the engine while we improve it. Once we make sure the engine is good enough to compute the limit of tasks it can auto-batch by itself, we will remove this parameter, and we don't want this to be a breaking change.

🗣️  **You are welcome to give feedback about the score details or ask any question on its usage; we are eager to collect feedback on the feature**

**Comment by gh2k** [comment]
Another issue beyond RAM usage here is the `no space left on device` problem that you can run into when ingesting large amounts of data in a short space of time: https://github.com/meilisearch/meilisearch/issues/2844 / https://github.com/meilisearch/meilisearch/issues/2765

(This is particularly an issue in Docker/k8s, where pointing `TMPDIR` at a mounted volume slows things down _a lot_ but root disk size can be limited.)

**Comment by mj520** [comment]
This is very useful and necessary、Looking forward to adding and default to the official version

**Reply by dureuill** [reply]
Thanks for your feedback. We'd really prefer if Meilisearch could determine the batch size itself dynamically, rather than the user having to specify it manually. We're still working on ways to improve the performance of indexing operations.
